python/src/trezorlib/cli/trezorctl.py

```diff
@@ -181,20 +188,25 @@ def cli_main(
         except ValueError:
             raise click.ClickException(f"Not a valid session id: {session_id}")
 
-    ctx.obj = TrezorConnection(path, bytes_session_id, passphrase_on_host)
+    ctx.obj = TrezorConnection(path, bytes_session_id, passphrase_on_host, script)
 
 
 # Creating a cli function that has the right types for future usage
 cli = cast(TrezorctlGroup, cli_main)
 
 
 @cli.resultcallback()
-def print_result(res: Any, is_json: bool, **kwargs: Any) -> None:
+def print_result(res: Any, is_json: bool, script: bool, **kwargs: Any) -> None:
     if is_json:
         if isinstance(res, protobuf.MessageType):
-            click.echo(json.dumps({res.__class__.__name__: res.__dict__}))
+            res_dict = protobuf.to_dict(res, hexlify_bytes=True)
+            click.echo(json.dumps({res.__class__.__name__: res_dict}))
         else:
-            click.echo(json.dumps(res, sort_keys=True, indent=4))
+            # Printing the result on one line for scripts, with indent for CLI users
+            if script:
+                click.echo(json.dumps(res))
+            else:
+                click.echo(json.dumps(res, sort_keys=True, indent=4))
```

Comment:
this seems complicated

IMHO the flow should be:
```
if isinstance(MessageType):
    res = protobuf.to_dict(...)

if script:
    json.dumps(res)  # no newlines
else:
   json.dumps(res, indent=4)  # pretty print
```

Reply:
Done like this in e5670a4